File: test/fixtures/testsuites/fakesubmission_test.hs
import Test.HUnit (Assertion, (@=?), runTestTT, Test(..), Counts (failures, errors))
import Control.Monad (void)
import System.Exit (ExitCode (ExitSuccess, ExitFailure), exitWith)
import FakeSubmission ( thisIsTrue )

testCase :: String -> Assertion -> Test
testCase label assertion = TestLabel label (TestCase assertion)

test_thisPasses :: Assertion
test_thisPasses = True @=? thisIsTrue

fakeSubmissionTests :: [Test]
fakeSubmissionTests = [ testCase "this passes" test_thisPasses ]

testsPass :: Counts -> Bool
testsPass testcounts = and [failures testcounts == 0,
                            errors   testcounts == 0]

exitStatus :: Counts -> ExitCode
exitStatus testcounts
  | testsPass testcounts = ExitSuccess
  | otherwise            = ExitFailure 1

main :: IO ()
main = do
  result <- runTestTT (TestList failingSubmissionTests)
  exitWith (exitStatus result)

File: test/fixtures/testsuites/fake_submission_test.py
try:
    import fake_submission
except ImportError:
    raise SystemExit('Could not find fake_submission.py. Does it exist?')

import unittest

class FakeSubmissionTests(unittest.TestCase):

    def test_this_passes(self):
        self.fake_submission = fake_submission.FakeSubmission()
        self.assertTrue(
            self.fake_submission.this_is_true()
        )

if __name__ == '__main__':
    unittest.main()
